### ISSUE TITLE
Targets/Modules Documentation

### ISSUE THREAD

**Issue by rathbuna** [EricZimmerman/KapeFiles#347]
For anyone looking to contribute to the documentation piece of this project, every Target/Module without any documentation currently has "N/A" listed under the commented Documentation heading. Feel free to search the file contents of all Targets and Modules for that string and contribute as you see fit. 

Any currently documented Targets and Modules are always available to have further documentation provided that isn't already. 

The endgame for this is to ensure Targets and Modules can serve to be educational in nature in addition to their primary functions. 

If anyone wants to learn how to contribute to GitHub, please reach out and I'm happy to walk you through.

**Comment by Karneades** [comment]
Should the version number be increased if documentation changes? Same question if only name has changed inside a compound, e.g. change RemoteAccess to RemoteAdmin in KapeFiles.tkape.

**Comment by rathbuna** [comment]
I personally consider those lateral/organizational moves. If the content of the Target or Module is progressed, then it makes sense to iterate the version number. Just my take on it. Anything that happens in the comments section doesn't change how the program works. It's merely just for educational/referential purposes.

**Comment by Karneades** [comment]
so, they will not get updated with a --sync or the like and only be integrated with the next release/version?

**Comment by rathbuna** [comment]
If there are any differences between what's in this KapeFiles repo and what you have locally on your system when you run --sync, it will grab those new files/changes at any time. Even if it's just a spacing issue or a removed/added comma, it'll grab the new file unless the modified file falls under the .gitignore, as I understand it.

**Comment by rathbuna** [comment]
> 
> 
> thanks for clarification

Yep no problem. I'm not an expert on the .gitignore stuff at this time, but everything else I said is how it works. Same with the EVTXECmd Maps, etc. The smallest changes will be synced when --sync is ran.

**Comment by Karneades** [comment]
Please leave the issue open for the documentation issue, don't you think?

**Comment by rathbuna** [comment]
Oh yeah, good point. My bad!

**Comment by EricZimmerman** [comment]
updates happen based on the SHA-1 of the file, NOT the version #.

so if ANYTHING changes that causes the sha-1 to change, its considered an update

**Comment by Karneades** [comment]
What documentation is preferred in compounds? A summary of the individual targets/modules documentation? Or should we omit adding a duplicated documentation to the compounds but refer to the documentation in the targets itself? In some compounds the documentation only contains "how to contribute to this compound" sentence.

**Comment by rathbuna** [comment]
Yeah I think duplicating documentation can get messy after a while, especially with some of the more complex Targets. Those "how to contribute" lines were more just pointers on how to expand those Targets, which there's plenty of room for that. I'm not quite sure what the best answer is but I don't think its duplication of effort by adding a bunch of lines of Documentation that already exist elsewhere. What are your thoughts?  

I'd like to keep the tips on contributing because on the off chance someone reads that sentence and acts on it, then the whole community benefits.

**Comment by Karneades** [comment]
Yes, we should omit duplicating documentation. Maybe for some compounds, we can just refer to the individual files using a generic sentence like "For documentation, please see the individual (t|m)kape files."

**Comment by EricZimmerman** [comment]
compound files are kind of self explanatory IMO. they just point to other, "real" files, which will then contain the documentation about what they are doing.
